src/sentencepiece_tokenizer.cc

```diff
@@ -37,7 +37,13 @@ class SentencePieceTokenizer : public Tokenizer {
 
   std::string IdToToken(int32_t id) final { return sentence_piece_.IdToPiece(id); }
 
-  int32_t TokenToId(const std::string& token) final { return sentence_piece_.PieceToId(token); }
+  int32_t TokenToId(const std::string& token) final {
+    int32_t id = sentence_piece_.PieceToId(token);
+    if (id == sentence_piece_.unk_id()) {
+      return -1;
+    }
+    return id;
+  }
```

Comment:
Shall we still align with sentencepiece and still return unk when the token is not found?

Reply:
Hello @Ubospica. Thanks for your comment!
I am not sure about current circumstances with sentencepiece, but I just wanted to enable `tokenToId` in web.
As I can see in the following description of `Tokenizer.TokenToId()`, -1 would be returned when it could not find the corresponding id.
https://github.com/mlc-ai/tokenizers-cpp/blob/5de6f656c06da557d4f0fb1ca611b16d6e9ff11d/include/tokenizers_cpp.h#L65-L68
The mentioned function is implementing that virtual method, so I wanted to make it as that.
(For HFTokenizer, it is already covered as I saw.)

As I understood, UNK token from sentencepiece tokenizer could be sth other than -1 due to the configuration. So I added explicit check with `sentence_piece_ .unk_id()`. But as I mentioned in the first comment, I am not sure this logic is really consistent considering practical uses, so you can suggest me to modify or revert that part to align it.

Reply:
Hello @grf53, thank you for your explanation! 

I think we should still align with Huggingface tokenizers, which means returning unk_token instead of -1 when the token does not exist. Additionally, please add tests to ensure that unk_token is returned for non-existent tokens.

For the current implementation of HFTokenizer, we will later modify it to return unk_token instead of -1.  

Thanks again for your contribution!